lib/solve/ruby_solver.rb

```diff
@@ -159,9 +162,29 @@ def search_for(dependency)
     # Callback required by Molinillo
     # @return [Boolean]
     def requirement_satisfied_by?(requirement, activated, spec)
-      requirement.constraint.satisfies?(spec.version)
+      version = spec.version
+      return false unless requirement.constraint.satisfies?(version)
+      shared_possibility_versions = possibility_versions(requirement, activated)
```

Comment:
molinillo 0.6 allows us to check all the vertices of the dependency graph at once, letting us resolve more quickly.